rfc/0003-policy-signing.md

```diff
@@ -0,0 +1,440 @@
+|              |                                  |
+| :----------- | :------------------------------- |
+| Feature Name | Policy Signing                   |
+| Start Date   | 21 Jan 2022                      |
+| Category     | Security                         |
+| RFC PR       | [#147](https://github.com/kubewarden/kubewarden-controller/pull/147)  |
+
+# Summary
+[summary]: #summary
+
+> Brief (one-paragraph) explanation of the feature.
+
+We want to provide a way to verify the integrity and authenticity of our policies, so that end-users can trust them.
+
+To accomplish that, we will provide ways to sign and verify policies.
+
+
+# Motivation
+[motivation]: #motivation
+
+> - Why are we doing this?
+> - What use cases does it support?
+> - What is the expected outcome?
+>
+> Describe the problem you are trying to solve, and its constraints, without coupling them too closely to the solution you have in mind. If this RFC is not accepted, the motivation can be used to develop alternative solutions.
+
+Our policies are binary blobs, that makes hard to ensure the policy we're about to run
+matches with the source code tagged inside of a git repository.
+
+This is what we want to achieve with this proposal:
+
+* policy authors: have a way to sign their policies, to attest their provenance.
+* end users: have a way to verify the integrity of a policy that they want to load inside of their cluster. Allow them to execute only policies they trust.
```

Comment:
Wouldn't that be a policy in itself. End users shouldn't be bothered about the internals. They just "trust this source/author" and rely on tools/ui to honor this trust.

Reply:
We also need to split "end user" into (at least two) sub-categories
- cluster administrators - they set policies and ultimately decide on the rules workloads (and policies) have to obey.
- cluster users - they want their workload to run and usually follow the install instructions says "download and run this binary". Policies, signing, etc. just get into their way.

Reply:
At the end of the day, none of them (authors, admins, users) will care about signing. Authors want their software to be easily deployed, admins want less administrative work, and users want to run software. It's the CIOs and CSO's of this world, asking for secure software chains.

That said, signing must be unobtrusive, ideally invisible, to get accepted.